.circleci/config.yml

```diff
@@ -0,0 +1,159 @@
+executors:
+  ruby_node:
+    docker:
+      - image: 'circleci/ruby:2.7.0-node-browsers'
+        environment:
+          - RAILS_ENV: test
+          - PGHOST: 127.0.0.1
+          - PGUSER: root
+      - image: circleci/node:12.16.1-browsers
+        environment:
+          - NODE_OPTIONS: --max_old_space_size=4096
+    working_directory: ~/tmp
+aliases:
+  - &attach_workspace
+    attach_workspace:
+      at: ~/tmp
+  - &restore_cache
+    restore_cache:
+        name: Restore Cache
+        keys:
+            - yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
+            - yarn-packages-{{ .Branch }}
+            - yarn-packages-master
+            - yarn-packages-
+            - v1-dependencies-{{ checksum "Gemfile.lock" }}
+            - v1-dependencies-
+jobs:
+  build:
+    executor: ruby_node
+    steps:
+      - checkout
+      - *attach_workspace
+      - *restore_cache
+      - run:
+          name: install dependencies
+          command: |
+            bundle install --deployment --path vendor/bundle
+      - save_cache:
+          paths:
+            - ./bundle
+          key: v1-dependencies-{{ checksum "Gemfile.lock" }}
+      - run:
+          name: install yarn dependencies
+          command: yarn install --check-files
+      - save_cache:
+          name: Save Yarn Package Cache
+          key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          paths:
+            - node_modules/
+      - run: bundle exec rake db:setup
+      - run: bundle exec rake db:test:prepare
+      - run: bundle exec rake assets:precompile
+      - run:
+          name:  Download cc-test-reporter
+          command: |
+            mkdir -p tmp/
+            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./tmp/cc-test-reporter
+            chmod +x ./tmp/cc-test-reporter
+            ./tmp/cc-test-reporter before-build
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+  backend:
+    executor: ruby_node
+    steps:
+      - checkout
+      - *attach_workspace
+      - *restore_cache
+      - run: bundle --path vendor/bundle
+      - run: ls tmp
+      - run: bundle exec rubocop
+      - run: bundle exec rake spec
+      - run: ls .
+      - persist_to_workspace:
+          root: .
+          paths:
+            - coverage
+  frontend:
+    executor: ruby_node
+    steps:
+      - checkout
+      - *attach_workspace
+      - *restore_cache
+      - run: yarn lint
+      - run: yarn unit
+      - run: ls .
+      - persist_to_workspace:
+          root: .
+          paths:
+            - coverage
+  e2e:
+    executor: ruby_node
+    steps:
+      - checkout
+      - *attach_workspace
+      - *restore_cache
+      - puppeteer/install
+      - run:
+          name: install yarn dependencies
+          command: yarn install --check-files
+      - run: bundle --path vendor/bundle
+      - run: bundle exec rake assets:precompile
```

Comment:
Now that we are persisting state correctly across jobs, we probably don't need to re-install packages, right?

Reply:
I'll give it a try